src/schema.ts

```diff
@@ -379,7 +380,7 @@ class QueryParser {
 }
 
 
-class Schema {
+class Schema<T> {
   paths: Record<string, SchemaType<any>> = {};
   statics: Record<string, (...args: any[]) => any> = {};
```

Comment:
```suggestion
  statics: Record<string, (this: T, ...args: any[]) => any> = {};
```

Is it possible to improve the types here as well? This could achieve stricter type checking.